submarine-commons/commons-rpc/src/main/proto/SubmarineServerProtocol.proto

```diff
@@ -0,0 +1,129 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+
+//   http://www.apache.org/licenses/LICENSE-2.0
+
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+syntax = "proto3";
+
+option java_multiple_files = true;
+option java_package = "org.apache.submarine.commons.rpc";
+option java_outer_classname = "SubmarineServerRpc";
+
+// Interface exported by the server.
+service SubmarineServerProtocol {
+  // Submit a job to submarine server.
+  rpc SubmitJob(ParameterProto) returns (ApplicationIdProto) {}
+
+  // test rpc
+  rpc TestRpc(ParametersHolderProto) returns (ApplicationIdProto) {}
+
+  // A server-to-client streaming RPC.
+  //
+  // Obtains the Features available within the given Rectangle.  Results are
+  // streamed rather than returned at once (e.g. in a response message with a
+  // repeated field), as the rectangle may cover a large area and contain a
+  // huge number of features.
+  // rpc ListFeatures(Rectangle) returns (stream Feature) {}
+
+  // A client-to-server streaming RPC.
+  //
+  // Accepts a stream of Points on a route being traversed, returning a
+  // RouteSummary when traversal is completed.
+  // rpc RecordRoute(stream Point) returns (RouteSummary) {}
+
+  // A Bidirectional streaming RPC.
+  //
+  // Accepts a stream of RouteNotes sent while a route is being traversed,
+  // while receiving other RouteNotes (e.g. from other users).
+  // rpc RouteChat(stream RouteNote) returns (stream RouteNote) {}
+}
+
+message ParametersHolderProto {
+  int32 helloworld = 1;
+}
+
+message ParameterProto {
```

Comment:
I was thinking what's the relationship between this parameter proto message and the submarine server REST job spec. Would it be a good idea to use this proto as a source of truth for all jobs? Which means, the current REST job spec will be used by the workbench, and when submarine client submit job with RPC protocal, the job spec will be translated to this ParameterProto. When submarine client submit job with REST protocal, the job spec will be sent directly to submarine server.

Reply:
Yes, we currently need two protocols to support the workbench and client respectively. 
My suggestion is not to stop the action and continue to promote the development of the project. 
We will reconsider this issue later.